nixos/modules/contracts/secret.nix

```
@@ -0,0 +1,123 @@
+# This "secret" contracts is an example and should be merged with the Vars initiative.
+{ lib, ... }:
+let
+  inherit (lib) mkOption;
+  inherit (lib.types) str path;
+in
+{
+  contracts.secret = {
+    meta = {
+      maitainers = [ lib.maintainers.ibizaman ];
+      description = ''
+        Secret contract where a consumer requests a secret
+        and a provider provides it at runtime at a given file path.
```

```suggestion
        and a provider determines statically the file path where the secret will be provided at runtime.
```
More accurate, but please double check

---

`s/statically/at evaluation time` ?
Also would "at activation time" instead of at "runtime" be more accurate too?